src/families/solana/test-dataset.ts

```diff
@@ -0,0 +1,572 @@
+import {
+  AmountRequired,
+  InvalidAddress,
+  InvalidAddressBecauseDestinationIsAlsoSource,
+  NotEnoughBalance,
+  RecipientRequired,
+} from "@ledgerhq/errors";
+import {
+  SolanaAccountNotFunded,
+  SolanaAddressOffEd25519,
+  SolanaMemoIsTooLong,
+  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+  SolanaRecipientAssociatedTokenAccountWillBeFunded,
+  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+  SolanaTokenAccountHoldsAnotherToken,
+} from "./errors";
+import {
+  encodeAccountIdWithTokenAccountAddress,
+  MAX_MEMO_LENGTH,
+} from "./logic";
+import createTransaction from "./js-createTransaction";
+import { compact } from "lodash/fp";
+import { assertUnreachable } from "./utils";
+import { getEnv } from "../../env";
+
+// do not change real properties or the test will break
+const testOnChainData = {
+  //  --- real props ---
+  unfundedAddress: "7b6Q3ap8qRzfyvDw1Qce3fUV8C7WgFNzJQwYNTJm3KQo",
+  // 0/0
+  fundedSenderAddress: "AQbkEagmPgmsdAfS4X8V8UyJnXXjVPMvjeD15etqQ3Jh",
+  fundedSenderBalance: new BigNumber(83389840),
+  // 1000/0
+  fundedAddress: "ARRKL4FT4LMwpkhUw4xNbfiHqR7UdePtzGLvkszgydqZ",
+  wSolSenderAssocTokenAccAddress:
+    "8RtwWeqdFz4EFuZU3MAadfYMWSdRMamjFrfq6BXkHuNN",
+  wSolSenderAssocTokenAccBalance: new BigNumber(7960720),
+  // 1000/0, mint - wrapped sol
+  wSolFundedAccountAssocTokenAccAddress:
+    "Ax69sAxqBSdT3gMAUqXb8pUvgxSLCiXfTitMALEnFZTS",
+  // 0/0
+  notWSolTokenAccAddress: "Hsm3S2rhX4HwxYBaCyqgJ1cCtFyFSBu6HLy1bdvh7fKs",
+  // ---  maybe outdated or not real, fine for tests ---
+  offEd25519Address: "6D8GtWkKJgToM5UoiByHqjQCCC9Dq1Hh7iNmU4jKSs14",
+  offEd25519Address2: "12rqwuEgBYiGhBrDJStCiqEtzQpTTiZbh7teNVLuYcFA",
+  feeCalculator: {
+    lamportsPerSignature: 5000,
+  },
+};
+
+const mainAccId = encodeAccountId({
+  type: "js",
+  version: "2",
+  currencyId: "solana",
+  xpubOrAddress: testOnChainData.fundedSenderAddress,
+  derivationMode: "solanaMain",
+});
+
+/* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+const wSolSubAccId = encodeAccountIdWithTokenAccountAddress(
+  mainAccId,
+  testOnChainData.wSolSenderAssocTokenAccAddress
+);
+
+const fees = (signatureCount: number) =>
+  new BigNumber(
+    signatureCount * testOnChainData.feeCalculator.lamportsPerSignature
+  );
+
+const zero = new BigNumber(0);
+
+const dataset: DatasetTest<Transaction> = {
+  implementations: [getEnv("MOCK") ? "mock" : "js"],
+  currencies: {
+    solana: {
+      scanAccounts: [scanAccounts1],
+      accounts: [
+        {
+          raw: makeAccount(testOnChainData.fundedSenderAddress),
+          /*
+          Solana integration is written in such a way that requires
+          prepareTransaction to be called before any meaningfull status
+          can be calculated. The general bridge tests do not run prepareTransaction
+          on all tests, so skip them in the general runner, but make alternatives for them.
```

Comment:
Yes, indeed I considered doing the command & validation in `js-getTransactionStatus` first. 
Doing the integration I was guided by ledger dev docs: https://developers.ledger.com/docs/coin/live-common/
When there was no answer in the dev docs I checked the implementations of other coins.
So, I decided to move command & validation to `js-prepareTransaction` because it has much higher time budget from product point of view. The doc says it can be `1-10 s` and `js-getTransactionStatus` only `0-2 s`. 
Another thing is `js-prepareTransaction` mainly maps ui model to a command. Validation is in there because sometimes it's not possible to do the mapping since, like incorrect address, etc.

Reply:
Makes a lot of sense, I think we should even take this in consideration for future evolution of our framework. Thanks.